File: python/7 kyu/196.py
# https://www.codewars.com/kata/5a3e1319b6486ac96f000049/train/python

# Passed

def pairs(arr):
    consecutive_pair_count = 0
    for idx in range(0, len(arr), 2):
        if idx + 1 == len(arr):
            break # last item
        
        if abs(arr[idx + 1] - arr[idx]) == 1:
            consecutive_pair_count += 1
            
    return consecutive_pair_count

output = pairs([21, 20, 22, 40, 39, -56, 30, -55, 95, 94])
print(output)

File: nodejs/7 kyu/196.js
// https://www.codewars.com/kata/52761ee4cffbc69732000738/train/javascript

// Passed

function goodVsEvil(good, evil) {
  let goodSide = [1, 2, 3, 3, 4, 10];
  let evilSide = [1, 2, 2, 2, 3, 5, 10];
  let goodSidePower = good
    .split(" ")
    .map((forceCount, forceIdx) => Number(forceCount) * goodSide[forceIdx])
    .reduce((prev, curr) => prev + curr, 0);
  let evilSidePower = evil
    .split(" ")
    .map((forceCount, forceIdx) => Number(forceCount) * evilSide[forceIdx])
    .reduce((prev, curr) => prev + curr, 0);
  let result;
  if (goodSidePower === evilSidePower) {
    result = "Battle Result: No victor on this battle field";
  } else {
    result =
      goodSidePower > evilSidePower
        ? "Battle Result: Good triumphs over Evil"
        : "Battle Result: Evil eradicates all trace of Good";
  }

  return result;
}

const output = goodVsEvil("0 0 0 0 0 10", "0 1 1 1 1 0 0");
console.log(output);